modules/llamacpp_model.py

```diff
@@ -84,12 +84,56 @@ def generate(self, prompt, state, callback=None):
             ]) if state['ban_eos_token'] else None,
         )
```

Comment:
Member function `Llama.create_completion()` [takes a `stop` parameter](https://github.com/abetlen/llama-cpp-python/blob/main/llama_cpp/llama.py#L785). Is there a reason we don't want to use it?

Reply:
Maybe not, I'm not familiar with llama.cpp but I guess we should use `stop` or `stopping_criteria` indeed

I just wrote the logic for `exllama` then copied it over to `llama.cpp`

I will try at a later time, many thanks to remind me

Reply:
I have tried both webui and api
It worked very well and I updated it up
Thanks again